hypershift-operator/main.go

```diff
@@ -253,16 +254,35 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		}
 	}
 
+	// The mgr and therefore the cache is not started yet, thus we have to construct a client that
```

Comment:
We should skip this section if CRDs for ImageContentSourcePolicy or ImageDigestMirrorSet are not present in the mgmt cluster (you can detect them via capabilities above). Otherwise we'll fail to start on a plain kube mgmt cluster.